lib/matplotlib/backends/backend_agg.py

```diff
@@ -625,13 +625,30 @@ def print_jpg(self, filename_or_obj, *args, **kwargs):
 
         # add TIFF support
         def print_tif(self, filename_or_obj, *args, **kwargs):
+            """
+            Other Parameters
+            ----------------
+            compressed : bool
+                If present, indicates that this image
+                should be stored using the LZW compression algorithm.
+            """
             buf, size = self.print_to_buffer()
             if kwargs.pop("dryrun", False):
                 return
             image = Image.frombuffer('RGBA', size, buf, 'raw', 'RGBA', 0, 1)
             dpi = (self.figure.dpi, self.figure.dpi)
-            return image.save(filename_or_obj, format='tiff',
-                              dpi=dpi)
+            #add TIFF compression support
+            compressed = kwargs.pop("compressed", False)
+            if compressed:
+                libtiff_original_value = TiffImagePlugin.WRITE_LIBTIFF
```

Comment:
Sure, I'll add the test part. 

I'd say it is not OK to "simplify" because you may use the flag somewhere else and remain confused about it not getting back to its original state once finished saving the image.